engine/apps/alerts/tasks/notify_ical_schedule_shift.py

```diff
@@ -183,160 +26,137 @@ def notify_ical_schedule_shift(schedule_pk):
         return
 
     if schedule.organization.slack_team_identity is None:
-        task_logger.info(f"Trying to notify ical schedule shift with no slack team identity {schedule_pk}")
+        task_logger.info(
+            f"Trying to notify ical schedule shift with no slack team identity {schedule_pk}, "
+            f"organization {schedule.organization_id}"
+        )
         return
     elif schedule.organization.deleted_at:
-        task_logger.info(f"Trying to notify ical schedule shift from deleted organization {schedule_pk}")
+        task_logger.info(
+            f"Trying to notify ical schedule shift from deleted organization {schedule_pk}, "
+            f"organization {schedule.organization_id}"
+        )
         return
 
+    task_logger.info(f"Notify ical schedule shift {schedule_pk}, organization {schedule.organization_id}")
+
     MIN_DAYS_TO_LOOKUP_FOR_THE_END_OF_EVENT = 3
 
     now = datetime.datetime.now(timezone.utc)
-    # get list of iCalendars from current iCal files. If there is more than one calendar, primary calendar will always
-    # be the first
-    current_calendars = schedule.get_icalendars()
 
-    current_shifts = {}
-    # expected current_shifts structure:
-    # {
-    #   some uid: {
-    #       "users": [users pks],
-    #       "start": event start date,
-    #       "end": event end date,
-    #       "all_day": bool if event has all-day type,
-    #       "priority": priority level,
-    #       "priority_increased_by": min priority level of primary calendar,  (for primary calendar event it is 0)
-    #   },
-    # }
-
-    # Current_user dict exists because it's bad idea to serialize User objects.
-    # Instead users' pks are stored in db for calculation related to shift diff.
-    # When it is needed to pass shift's user (e.g. in def get_report_blocks_ical())
-    # we take users from current_users{} by shift uuid and replace users' pk
-    current_users = {}
-
-    overrides_priority = 0
-    for calendar in current_calendars:
-        if calendar is not None:
-            current_shifts_result, current_users_result = get_current_shifts_from_ical(
-                calendar,
-                schedule,
-                overrides_priority,
+    current_shifts = schedule.final_events(now, now, False, False)
+
+    prev_shifts = json.loads(schedule.current_shifts) if not schedule.empty_oncall else []
+    prev_shifts_updated = False
+    # convert prev_shifts to new events format for compatibility with the previous version of this task
```

Comment:
It will be nice to leave a task for later to drop this conversion. Optional, if you think it makes sense, I would separate this "translation" in a helper function in this module (so it is simpler to follow the task logic ignoring this compatibility thing)